.github/workflows/ruby.yml

```diff
@@ -337,13 +340,15 @@ jobs:
       ARROW_WITH_LZ4: OFF
       ARROW_WITH_OPENTELEMETRY: OFF
       ARROW_WITH_SNAPPY: ON
-      ARROW_WITH_ZLIB: OFF
+      ARROW_WITH_ZLIB: ON
       ARROW_WITH_ZSTD: ON
       BOOST_SOURCE: BUNDLED
```

Comment:
Can we remove this?

Reply:
Yes this is implied by ARROW_WITH_FLIGHT so doesn't need to be set explicitly, I'll remove it

Reply:
Oh sorry I realise now that this comment is about the BOOST_SOURCE variable, not the ZLIB setting. Yes this should be removed, boost is coming from vcpkg so this shouldn't have any effect.